addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/config/SleepIQCloudConfiguration.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sleepiq.config;
+
+/**
+ * Configuration class for the SleepIQ cloud.
+ *
+ * @author Gregory Moyer - Initial contribution
+ */
+public class SleepIQCloudConfiguration {
+
+    public static final String USERNAME = "username";
+    public static final String PASSWORD = "password";
+    public static final String POLLING_INTERVAL = "pollingInterval";
+    public static final String LOGGING = "logging";
```

Comment:
I think it would be best if I described what the library is doing first. It is using the version of Jersey that is available in OH/ESH which has a LoggingFilter class. This is the old logging filter implementation that does not allow for selecting a logging level - it logs everything at the INFO level. When the logging configuration option is set to true, the filter is added to the Jersey client using the package name of the library and all request/responses are subsequently logged.

Therefore, if the root logger in OH is set to INFO or lower, turning on this flag in the binding will cause the log messages to be printed using the org.syphr.sleepiq.impl.SleepIQImpl logger. Of course, setting that specific logger (or any of its parents) to INFO or lower would also start the logging if the flag is turned on. I'm not sure what the default logging level is in OH, but since INFO is a common default, not spamming the logs was the motivation for this parameter in the first place.

If I changed the binding to use ```logger.isDebugEnabled()``` the logging will not turn on simply by setting the log level. The binding would need to detect the log level change and reinitialize the cloud service client or simply the entire Thing. I don't know of an elegant way to listen for log level changes, but the configuration option on the Thing provides exactly that functionality.

In newer versions of Jersey, the logging filter allows the level to be set to something more expected (e.g. debug or trace). If that were available, the configuration option would not be needed. As it stands, I don't see a good way to avoid spamming the logs with HTTP requests or turning off the logging entirely without having the configuration option.